lib/active_admin/base_controller/authorization.rb

```diff
@@ -82,7 +82,11 @@ def active_admin_authorization
       def active_admin_authorization_adapter
         adapter = active_admin_namespace.authorization_adapter
         if adapter.is_a? String
-          ActiveSupport::Dependencies.constantize adapter
+          if ActiveSupport::Dependencies.respond_to?(:constantize)
+            ActiveSupport::Dependencies.constantize(adapter)
+          else
+            adapter.constantize
+          end
```

Comment:
you may want to start reviewing this. There is also code duplication

I don't know if there are side effects and if it a good idea to have a single constantizer helper defined in the ActiveAdmin module so that both AuthorizationController and Resource can use it

Reply:
`String#constantize` has been available since Rails 3, so I think it's safe to use, and that removes the need for a helper.

Reply:
Thanks.

I've actually checked git blame and that method has been introduced in a commit 10 years ago 572dbf1ffafcdc30d41aa2146b4a49ffb8b1c32d

Will remove the helper and use constantize

Reply:
@technicalpickles I've tried to remove `ActiveSupport::Dependencies.constantize(resource_name)`, but specs are failing now, even on 7.0

There is a difference in behavior

`rspec ./spec/unit/filters/active_filter_spec.rb:220 `

```
(byebug) resource_name
"::SuperPost"
(byebug) ActiveSupport::Dependencies.constantize(resource_name)
#<Class:0x00007f9cd686fdb0>(id: integer, title: string, body: text, published_date: date, author_id: integer, position: integer, custom_category_id: integer, starred: boolean, foo_id: integer, created_at: datetime, updated_at: datetime)
(byebug) resource_name.constantize
*** NameError Exception: uninitialized constant SuperPost

nil
```

Ref: https://github.com/activeadmin/activeadmin/runs/5028708103?check_suite_focus=true

Reply:
Spent a little time looking at how `constantize` works, and it seems that ActiveSupport::Dependencies.constantize is pretty closely tied to classic model. The zeitwerk mode only does the `constantize` like this, so ActiveSupport::Dependencies.constantize must be doing some extra work around loading.

I think that means adding a private helper to use `ActiveSupport::Dependencies.constantize` if available, or `adapter.constantize` otherwise.

Reply:
Oops, I missed this is happening on Rails 7 too 🤔

Reply:
> Oops, I missed this is happening on Rails 7 too 🤔

Hi @technicalpickles , sorry, I think that it may be the test itself

https://github.com/activeadmin/activeadmin/blob/1b6dd5c2c5722f896ca36c3fa447b4db63038446/spec/unit/filters/active_filter_spec.rb#L221

Reply:
@technicalpickles apologies, my fault. I didn't even check the spec definition. Now it is fixed

Aside of using jsbundling, tracked in #7264,  the only known issue is the `before_action` setting